src/main/scala/nl/knaw/dans/easy/pid/microservice/HazelcastService.scala

```diff
@@ -19,35 +19,25 @@ import com.hazelcast.Scala.client._
 import com.hazelcast.Scala.serialization
 import com.hazelcast.client.config.ClientConfig
 import com.hazelcast.core.HazelcastInstance
+import nl.knaw.dans.easy.pid.Service
 import org.apache.commons.daemon.{Daemon, DaemonContext}
 import org.slf4j.LoggerFactory
 
-class ServiceStarter extends Daemon {
-
+case class HazelcastService(conf: com.typesafe.config.Config) extends Service {
   val log = LoggerFactory.getLogger(getClass)
   var hz: HazelcastInstance = _
 
-  def init(context: DaemonContext): Unit = {
-    log.info("Initializing pid-generator service ...")
-  }
-
-  def start(): Unit = {
-    log.info("Starting pid-generator service ...")
-
+  override def start(): Unit = {
+    log.info("Running as Hazelcast client ...")
     val conf = new ClientConfig()
     serialization.Defaults.register(conf.getSerializationConfig)
     hz = conf.newClient()
-
     PidGeneratorService.run(hz) // can't pass this implicitly since `hz` is a variable
   }
 
-  def stop(): Unit = {
-    log.info("Stopping pid-generator service ...")
+  override def stop(): Unit = {
```

Comment:
**A quick word of preference/style here, which is to be taken as a personal note!**

you don't really need the `override` keyword here, as you are *implementing* this function here rather than *overriding* it. To me the `override` means literally *override the implementation* rather than *implement it for the first time*. Got me confused for a moment :-) On the other hand, it makes it more clear that this method is not original here, but that it is enforced by some other interface/trait/(abstract) class. So there is to say something for both sides!